test-validator/src/lib.rs

```diff
@@ -1158,6 +1183,61 @@ impl TestValidator {
         }
     }
 
+    /// programs added to genesis ain't immediately usable. Actively check "Program
+    /// is not deployed" error for their availibility.
+    async fn wait_for_upgradeable_programs_deployed(
+        &self,
+        upgradeable_programs: &[&Pubkey],
+        payer: &Keypair,
+    ) {
+        let rpc_client = nonblocking::rpc_client::RpcClient::new_with_commitment(
+            self.rpc_url.clone(),
+            CommitmentConfig::processed(),
+        );
+
+        const MAX_ATTEMPTS: u64 = 10;
+        let mut attempt = 0;
+        loop {
+            let blockhash = rpc_client.get_latest_blockhash().await.unwrap();
+            let mut all_deployed = true;
+            for program_id in upgradeable_programs {
```

Comment:
If the program_id has been successful in previous iterations, we don't have to resend

Reply:
9ea07b5851f98d20e981e600efe97ea002c7a08a has it